include/coap2/coap_forward_decls.h

```diff
@@ -45,47 +41,65 @@ typedef struct coap_pdu_t coap_pdu_t;
 typedef struct coap_context_t coap_context_t;
 
 /*
- * typedef all the opaque structures that are defined in coap_*_internal.h
+ * forward declare and typedef all the opaque structures that are defined in
+ * coap_*_internal.h
  */
 
 /* ************* coap_cache_internal.h ***************** */
 
-/**
- * Cache Entry information
+/*
+ * Cache Entry information.
  */
+struct coap_cache_entry_t;
+struct coap_cache_key_t;
+
 typedef struct coap_cache_entry_t coap_cache_entry_t;
 typedef struct coap_cache_key_t coap_cache_key_t;
 
 /* ************* coap_block_internal.h ***************** */
 
-/**
-* Block handling information.
-*/
+/*
+ * Block handling information.
+ */
+struct coap_lg_xmit_t;
+struct coap_lg_crcv_t;
+struct coap_lg_srcv_t;
+
 typedef struct coap_lg_xmit_t coap_lg_xmit_t;
 typedef struct coap_lg_crcv_t coap_lg_crcv_t;
-
 typedef struct coap_lg_srcv_t coap_lg_srcv_t;
 
 /* ************* coap_cache_internal.h ***************** */
 
-/**
- * Resource information
+/*
+ * Resource information.
  */
+struct coap_attr_t;
+struct coap_resource_t;
+
 typedef struct coap_attr_t coap_attr_t;
 typedef struct coap_resource_t coap_resource_t;
 
 /* ************* coap_session_internal.h ***************** */
 
-/**
- * Abstraction of virtual endpoint that can be attached to coap_context_t.
+/*
+ * Session information.
  */
+struct coap_addr_hash_t;
```

Comment:
With the following typedefs, the forward declarations for the structs are not required.

Reply:
Something I just learnt - thanks.  Code changes pushed.  Will do the same in the other obfuscate PRs.